core/server/services/webhooks/trigger.js

```diff
@@ -23,7 +24,8 @@ function makeRequests(webhooksCollection, payload, options) {
             body: reqPayload,
             headers: {
                 'Content-Length': Buffer.byteLength(reqPayload),
-                'Content-Type': 'application/json'
+                'Content-Type': 'application/json',
+                'X-Ghost-Secret': webhookSecret
```

Comment:
Has the secret behaviour been specced/discussed anywhere? The header looks fine I'm just wondering how it fits with how other services handle secrets/signatures.

Eg, https://developer.github.com/webhooks/securing/ and https://www.netlify.com/docs/webhooks/ both talk about using the secret to generate a signature so that you can compare secrets without the secret actually being transmitted. That seems like a better approach as it prevents MITM attacks.

I'm wondering if we should remove the secret field in the UI and hold off on adding a header until we have a fully thought through approach that we can document. Our initial use-case of triggering netlify rebuilds does not require a secret.

Reply:
@rishabhgrg confirmed, let's remove the header for now and circle back later with a more secure approach